docs/other/fontsobject.rst

```diff
@@ -29,24 +29,39 @@ FontsObject.allFonts
 
 **Description**
 
-The list of all the fonts currently available on your system, sorted by Font Family.
+The list of all the fonts currently available on your system.
+
+They are grouped into what is named a family group which are Arrays of :ref:`fontobject`.
+
+.. Naming and ordering::
+
+The Family Name of the group is simply the ``Font.familyName`` of any of the :ref:`fontobject` in the group.
+
+The Family Name in one font group is not guaranteed to have unique name compared to different font groups - the grouping is determined by a number of factors including the returned value of ``Font.technology`` and ``Font.writingScripts``.
+
+In addition, it is perfectly acceptable to have multiple fonts with the same PostScript name, though only one will have the same (PostScript name, Technology, Primary Writing Script) tuple. In the case of true duplicates, it is undefined which will be returned and which will be suppressed.
+
+The family groups and :ref:`fontobject` in the group are sorted accordingly to the setting in the ``Character Panel`` dropdown ``Show Font Names in English``. If set to true, the ``Font.familyName`` and ``Font.styleName`` property is used, otherwise the ``Font.nativeFamilyName`` and ``Font.nativeStyleName`` property is used.
+
+:ref:`fontobject` for which ``Font.isSubstitute`` returns true are always sorted to the end as individual family groups.
+
 
 **Type**
 
-Array of Arrays; read-only.
+Array of Arrays of :ref:`fontobject`; read-only.
```

Comment:
Prefer the use of labels for references so the sentences can read correctly while still linking to the Font object entry.

```suggestion
Array of Arrays of :ref:`Font objects <fontobject>`; read-only.
```